src/core/obj.js

```diff
@@ -135,6 +135,19 @@ var Dict = (function DictClosure() {
       return Promise.resolve(value);
     },
 
+    // Same as get(), but dereferences all elements if the result is an Array.
+    getArray: function Dict_getArray(key1, key2, key3) {
+      var value = this.get(key1, key2, key3);
+      if (!isArray(value)) {
+        return value;
+      }
+      var valueLen = value.length, valueArray = new Array(valueLen);
```

Comment:
I actually did that initially, but I had to change it since it doesn't work like you might think initially!
The problem is that just modifying `value` can actually modify the original `Dict` data, e.g. with the following code:
```javascript
xobj.dict.get('BBox');
xobj.dict.getArray('BBox');
xobj.dict.get('BBox')
```
You'd expect this to return (for the referenced PDF file):
```
Array [ 0, 0, 200, Object ]
Array [ 0, 0, 200, 50 ]
Array [ 0, 0, 200, Object ]
```
But with your suggestion, this is what *actually* happens:
```
Array [ 0, 0, 200, Object ]
Array [ 0, 0, 200, 50 ]
Array [ 0, 0, 200, 50 ]
```
---
However, I agree that this might not be immediately, clear. Would using e.g. `slice()` instead help with readability?